packages/3id-did-resolver/src/index.ts

```diff
@@ -88,13 +93,53 @@ export function wrapDocument(content: any, did: string): DIDDocument {
   return doc
 }
 
-// TODO - add backwards compatibility for 3IDv0
+const isLegacyDid = (didId: string): boolean => {
+  try {
+    new CID(didId) 
+    return true
+  } catch(e) {
+    return false
+  }
+}
+
+const getVersion = (query = ''): string | null => {
+  const versionParam = query.split('&').find(e => e.includes('version-id'))
+  return versionParam ? versionParam.split('=')[1] : null
+}
+ 
+const legacyResolve = async (ceramic: Ceramic, didId: string, version?: string): Promise<DIDDocument | null> => {
+  // todo could add opt to pass ceramic ipfs here instead when true
+  const legacyDoc = await LegacyResolver(didId)
+  if (!legacyDoc) return null
+  if (version === '0') return legacyDoc
+
+  const keyEntry = legacyDoc.publicKey.findIndex(e => e.id.endsWith('managementKey'))
+  const managementKey = legacyDoc.publicKey[keyEntry].ethereumAddress 
+  if (!managementKey) return null
+
+  try {
+    const content =  { metadata: { controllers: [`did:key:${managementKey}`], tags: ['3id'] } }
```

Comment:
We can't have a `did:key` that's an ethereumAddress. Therefore in IDW you can see that we use the `signingKey` as the `managementKey` for the genesis record.